src/zenml/integrations/kserve/steps/kserve_deployer.py

```diff
@@ -278,3 +317,154 @@ def kserve_model_deployer_step(
     )
 
     return service
+
+
+@step(enable_cache=False)
+def kserve_custom_model_deployer_step(
+    deploy_decision: bool,
+    config: KServeDeployerStepConfig,
+    context: StepContext,
+    model: ModelArtifact,
+) -> KServeDeploymentService:
```

Comment:
The custom code functionality could just be added to the existing `kserve_model_deployer_step` and have some code executed conditionally based on whether `custom_deploy_parameters` has been set or not. Some code duplication could have been avoided that way.